src/commands/functions-delete.ts

```diff
@@ -33,58 +34,65 @@ export default new Command("functions:delete [filters...]")
-        )}.`,
-        { exit: 1 }
-      );
-    }
+    try {
+      const [config, existingBackend] = await Promise.all([
+        functionsConfig.getFirebaseConfig(options),
+        backend.existingBackend(context),
+      ]);
+      await backend.checkAvailability(context, /* want=*/ backend.empty());
+      const appEngineLocation = functionsConfig.getAppEngineLocation(config);
 
-    const schedulesToDelete = existingBackend.schedules.filter((schedule) => {
-      functionsToDelete.some(backend.sameFunctionName(schedule.targetService));
-    });
-    const topicsToDelete = existingBackend.topics.filter((topic) => {
-      functionsToDelete.some(backend.sameFunctionName(topic.targetService));
-    });
+      const functionsToDelete = existingBackend.cloudFunctions.filter((fn) => {
+        const regionMatches = options.region ? fn.region === options.region : true;
+        const nameMatches = helper.functionMatchesAnyGroup(fn, filterChunks);
+        return regionMatches && nameMatches;
+      });
+      if (functionsToDelete.length === 0) {
+        throw new Error(
+          `The specified filters do not match any existing functions in project ${clc.bold(
+            context.projectId
+          )}.`
+        );
+      }
+
+      const schedulesToDelete = existingBackend.schedules.filter((schedule) => {
+        functionsToDelete.some(backend.sameFunctionName(schedule.targetService));
+      });
+      const topicsToDelete = existingBackend.topics.filter((topic) => {
+        functionsToDelete.some(backend.sameFunctionName(topic.targetService));
+      });
 
-    const deleteList = functionsToDelete
-      .map((func) => {
-        return "\t" + helper.getFunctionLabel(func);
-      })
-      .join("\n");
-    const confirmDeletion = await promptOnce(
-      {
-        type: "confirm",
-        name: "force",
-        default: false,
-        message:
-          "You are about to delete the following Cloud Functions:\n" +
-          deleteList +
-          "\n  Are you sure?",
-      },
-      options
-    );
-    if (!confirmDeletion) {
-      return utils.reject("Command aborted.", { exit: 1 });
+      const deleteList = functionsToDelete
+        .map((func) => {
+          return "\t" + helper.getFunctionLabel(func);
+        })
+        .join("\n");
+      const confirmDeletion = await promptOnce(
+        {
+          type: "confirm",
+          name: "force",
+          default: false,
+          message:
+            "You are about to delete the following Cloud Functions:\n" +
+            deleteList +
+            "\n  Are you sure?",
+        },
+        options
+      );
+      if (!confirmDeletion) {
+        throw new Error("Command aborted.");
+      }
+      return await deleteFunctions(
+        functionsToDelete,
+        schedulesToDelete,
+        topicsToDelete,
+        appEngineLocation
+      );
+    } catch (err) {
+      throw new FirebaseError(`Failed to delete functions ${err.message}`, {
```

Comment:
I'm wondering if we need err.message or if we should just set `original: err` in the options. Can you try that out and see if it renders correctly?

Reply:
It renders correctly and hides the actual error message under the `--debug` flag

Reply:
Do we want to hide potentially useful error messages under the debug flag? Spitting out the `err.message` can save customers some time debugging.